### PR TITLE
hv: inst_emul: clean the "Procedure has more than one exit point"

### PR DESCRIPTION
**Pull request by lifeix**
Misra C requires Function must have only 1 return entry.

Tracked-On: #861
Signed-off-by: Li, Fei1 <fei1.li@intel.com>

### CODE REVIEW

**Comment by lifeix** [comment]
@fyin1 Could u help to review ?

File: hypervisor/arch/x86/guest/instr_emul.c [hunk 1]
```diff
@@ -426,56 +426,59 @@ static void get_guest_paging_info(struct acrn_vcpu *vcpu, struct instr_emul_ctxt
...
-		}
+	if (type < 16U || type > 31U) {
+		ret = false;
+	} else {
+		if ((prot & PROT_READ) != 0U) {
+			/* #GP on a read access to a exec-only code segment */
+			if ((type & 0xAU) == 0x8U) {
+				ret = false;
+			}
+		} else if ((prot & PROT_WRITE) != 0U) {
```
Comment (dongyaozu) [inline hypervisor/arch/x86/guest/instr_emul.c]: if prot include both READ & WRITE, the semantics is changed

File: hypervisor/arch/x86/guest/instr_emul.c [hunk 2]
```diff
@@ -426,56 +426,59 @@ static void get_guest_paging_info(struct acrn_vcpu *vcpu, struct instr_emul_ctxt
...
 }
 
 static bool is_desc_valid(struct seg_desc *desc, uint32_t prot)
 {
+	bool ret = true;
 	uint32_t type;
 
 	/* The descriptor type must indicate a code/data segment. */
 	type = seg_desc_type(desc->access);
-	if (type < 16U || type > 31U) {
```
Comment (dongyaozu) [inline hypervisor/arch/x86/guest/instr_emul.c]: This is a bit defined variable and & 0x1f.    > 31U  never happen.

File: hypervisor/arch/x86/guest/instr_emul.c [hunk 3]
```diff
@@ -918,7 +928,8 @@ static void get_gva_si_nocheck(const struct acrn_vcpu *vcpu, uint8_t addrsize,
 static int32_t get_gva_di_check(struct acrn_vcpu *vcpu, struct instr_emul_vie *vie,
 		uint8_t addrsize, uint64_t *gva)
 {
-	int32_t ret;
+	bool inject_gp = false;
```
Comment (dongyaozu) [inline hypervisor/arch/x86/guest/instr_emul.c]: has_exception?  to align with other patches.

File: hypervisor/arch/x86/guest/instr_emul.c [hunk 4]
```diff
@@ -929,47 +940,50 @@ static int32_t get_gva_di_check(struct acrn_vcpu *vcpu, struct instr_emul_vie *v
...
-			goto exception_inject;
+			inject_gp = true;
 		}
 	} else {
 		if ((addrsize != 2U) && (addrsize != 4U)) {
-			goto exception_inject;
+			inject_gp = true;
 		}
 
 		if (!is_desc_valid(&desc, PROT_WRITE)) {
```
Comment (dongyaozu) [inline hypervisor/arch/x86/guest/instr_emul.c]: else here?
otherwise the semantics is different

File: hypervisor/arch/x86/guest/instr_emul.c [hunk 5]
```diff
@@ -929,47 +940,50 @@ static int32_t get_gva_di_check(struct acrn_vcpu *vcpu, struct instr_emul_vie *v
...
+		if (vie_calculate_gla(cpu_mode, CPU_REG_ES, &desc, val, addrsize, gva) != 0) {
+			inject_gp = true;
+		} else if (vie_canonical_check(cpu_mode, *gva) != 0) {
+			inject_gp = true;
+		} else {
+			err_code = PAGE_FAULT_WR_FLAG;
+			ret = gva2gpa(vcpu, *gva, &gpa, &err_code);
+			if (ret < 0) {
+				if (ret == -EFAULT) {
+					vcpu_inject_pf(vcpu, (uint64_t)gva, err_code);
```
Comment (dongyaozu) [inline hypervisor/arch/x86/guest/instr_emul.c]: inject_gp = true?

**Comment by acrnsi** [comment]
Ok to verify

**Comment by HuiHuangShi** [comment]
2265<br>No new violations to the coding guideline detected.

**Pull request merged**